src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -4053,12 +4039,12 @@ private void ReinferMethodAndVisitArguments(BoundCall node, TypeWithState receiv
             if (!receiverType.HasNullType)
             {
                 // Update method based on inferred receiver type.
-                method = (MethodSymbol)AsMemberOfType(receiverType.Type, method);
+                method = (MethodSymbol)AsMemberOfType(receiverType.Type!, method);
```

Comment:
Please mark `.HasNullType` and this site with `// use MemberNotNullWhen when available: https://github.com/dotnet/roslyn/issues/41964`.